### PR TITLE
New style/core some functions1

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/regular/ceil.hpp [hunk 1]
```diff
@@ -93,6 +108,6 @@ EVE_MAKE_CALLABLE(ceil_, ceil);
 #  include <eve/module/core/regular/impl/simd/arm/neon/ceil.hpp>
 #endif
 
-#if defined(EVE_INCLUDE_SVE_HEADER)
-#  include <eve/module/core/regular/impl/simd/arm/sve/ceil.hpp>
-#endif
+// #if defined(EVE_INCLUDE_SVE_HEADER)
```
Comment (jfalcou) [inline include/eve/module/core/regular/ceil.hpp:111]: Need to be put back

File: include/eve/module/core/regular/floor.hpp [hunk 2]
```diff
@@ -93,6 +108,6 @@ EVE_MAKE_CALLABLE(floor_, floor);
 #  include <eve/module/core/regular/impl/simd/arm/neon/floor.hpp>
 #endif
 
-#if defined(EVE_INCLUDE_SVE_HEADER)
-#  include <eve/module/core/regular/impl/simd/arm/sve/floor.hpp>
-#endif
+// #if defined(EVE_INCLUDE_SVE_HEADER)
```
Comment (jfalcou) [inline include/eve/module/core/regular/floor.hpp:111]: Need to be put back

File: include/eve/module/core/regular/impl/simd/x86/average.hpp [hunk 3]
```diff
@@ -9,67 +9,72 @@
...
+                                       C          const &cx,
+                                       O          const &opts,
+                                       wide<T, N> const &v,
+                                       wide<T, N> const &w) noexcept
+  requires x86_abi<abi_t<T, N>>
   {
-    auto src = alternative(cx, v, as<wide<T, N>> {});
-    auto m   = expand_mask(cx, as<wide<T, N>> {}).storage().value;
+    if constexpr( O::contains(upward2))
+      return average_(EVE_TARGETS(cpu_), cx, opts, v, w);
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/average.hpp:64]: should be behavior()

File: include/eve/module/core/regular/impl/simd/x86/ceil.hpp [hunk 4]
```diff
@@ -19,41 +19,48 @@
...
+      auto src = alternative(cx, v, as<wide<T, N>> {});
+      auto m   = expand_mask(cx, as<wide<T, N>> {}).storage().value;
+      if constexpr( C::is_complete )              return src;
+      else if constexpr( match(c, category::integer_)) return if_else(cx, v, src);
+      else if constexpr( c == category::float32x16 )   return _mm512_mask_ceil_ps(src, m, v);
+      else if constexpr( c == category::float64x8 )    return _mm512_mask_ceil_pd(src, m, v);
+      else if constexpr( match(c, category::float_))   return if_else(cx, eve::ceil(v), src);
+    }
+    else
+      return ceil_(EVE_TARGETS(cpu_), cx, o, v);
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/ceil.hpp:64]: behavior ?

File: include/eve/module/core/regular/impl/simd/x86/trunc.hpp [hunk 5]
```diff
@@ -17,51 +17,59 @@
...
+      else if constexpr( c == category::float64x4 )
+        return _mm256_mask_roundscale_pd(src, m, v, _MM_FROUND_TO_ZERO);
+      else if constexpr( c == category::float32x4 )
+        return _mm_mask_roundscale_ps(src, m, v, _MM_FROUND_TO_ZERO);
+      else if constexpr( c == category::float64x2 )
+        return _mm_mask_roundscale_pd(src, m, v, _MM_FROUND_TO_ZERO);
+      else return if_else(cx, v, src);
+    }
+    else
+      return trunc_(EVE_TARGETS(cpu_), cx, o, v);
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/trunc.hpp:73]: behavior ?

File: include/eve/module/core/regular/nearest.hpp [hunk 6]
```diff
@@ -8,10 +8,29 @@
...
-#include <eve/detail/overload.hpp>
+#include <eve/traits/overload.hpp>
+#include <eve/module/core/decorator/core.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct nearest_t : elementwise_callable<nearest_t, Options>
+  {
+    template<eve::integral_value T>
```
Comment (jfalcou) [inline include/eve/module/core/regular/nearest.hpp:19]: No logic in operator()

File: include/eve/module/core/saturated/core.hpp [hunk 7]
```diff
@@ -10,10 +10,8 @@
 
 #include <eve/module/core/saturated/add.hpp>
 #include <eve/module/core/saturated/convert.hpp>
-#include <eve/module/core/saturated/dec.hpp>
```
Comment (jfalcou) [inline include/eve/module/core/saturated/core.hpp:13]: Only those are removed ? We modified a lot of functions here, I woudl have expected more removals.

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/core/regular/ceil.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/ceil.hpp:111]: done

File: include/eve/module/core/regular/impl/simd/x86/ceil.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/ceil.hpp:64]: done

File: include/eve/module/core/regular/impl/simd/x86/trunc.hpp (hunk 5 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/trunc.hpp:73]: done

File: include/eve/module/core/saturated/core.hpp (hunk 7 above)
Comment (jtlap) [reply include/eve/module/core/saturated/core.hpp:13]: found 2 more

**Review by jfalcou — APPROVED** [review]

**Pull request merged**